useful_resources/sample_code/zoom_to_van.py

```diff
@@ -0,0 +1,71 @@
+# ### METADATA
+
+# Connectors: Zoom, VAN
+# Description: Adds Zoom attendees to VAN and applies an activist code.
+
+# ### CONFIGURATION
+
+# Set the configuration variables below or set environmental variables of the same name and
+# leave these with empty strings.  We recommend using environmental variables if possible.
+
+config_vars = {
+    # Zoom Authentication
+    "ZOOM_API_KEY": "",
+    "ZOOM_API_SECRET": "",
+    # Van Authentication
+    "VAN_API_KEY": ""
+}
+
+VAN_DB = "MyCampaign"  # one of: MyMembers, EveryAction, MyCampaign (not MyVoters)
+ACTIVIST_CODE_NAME = ""  # name of VAN activist code, which must be created manually in VAN
+ZOOM_MEETING_ID = ""
+MINIMUM_DURATION = 0  # filters out Zoom participants who stayed for less than minimum duration
+
+# ### CODE
+
+# Setup
+
+import os  # noqa: E402
+from parsons import VAN, Zoom  # noqa: E402
+
+# if variables specified above, sets them as environmental variables
+for name, value in config_vars.items():
+    if value.strip() != "":
+        os.environ[name] = value
+
+# Code
+
+zoom = Zoom()
+van = VAN(db=VAN_DB)
+
+# Gets participants from Zoom meeting
+participants = zoom.get_past_meeting_participants(ZOOM_MEETING_ID)
+filtered_participants = participants.select_rows(lambda row: row.duration > MINIMUM_DURATION)
+
+# Coalesce the columns into something VAN expects
+column_map = {"first_name": ["fn", "first", "firstname", "first name"],
+              "last_name": ["ln", "last", "lastname", "last name"],
+              "date_of_birth": ["dob", "date of birth", "birthday"],
+              "email": ["email address", "email_address"],
+              "street_number": ["street number", "street no.", "street no", "street #"],
+              "street_name": ["street name", "street"],
+              "phone": ["phone_number", "phone #", "phone_#", "phone no.", "phone no"],
+              "zip": ["zip5", "zipcode", "zip code"]}
+filtered_participants.map_and_coalesce_columns(column_map)
```

Comment:
This is an interesting pattern.. I wonder if we should think about moving some of these into Parsons.. like this could be our "Person map"..?

Reply:
There's a Slack thread from a few days ago where I proposed something vaguely like this, and Elyse seemed approving. It definitely feels like there's a couple things here that could be abstracted out.  I like the idea of maps, though they might need to be connector-specific (the keys in that dict are all the variable names accepted/required by find_person, upsert_person, etc **in the VAN connector**). The other element is this bit: `params = {col: participant[col] for col in column_map.keys() if participant[col]}` which is trying to make it easy to supply row data as parameters to a method call.